zrml/prediction-markets/src/lib.rs

```diff
@@ -907,12 +909,13 @@ mod pallet {
         pub fn reject_market(
             origin: OriginFor<T>,
             #[pallet::compact] market_id: MarketIdOf<T>,
+            reject_reason: RejectReason,
```

Comment:
Maybe it makes sense to use `Vec` here like in [pallet_bounties](https://github.com/paritytech/substrate/blob/0ee03277c33b6334ddba7434e014fa637dcb6107/frame/bounties/src/lib.rs#L816-L822).
The app that crafts the transaction can include a vector of arbitrary size anyways, only when the transaction is executed on-chain and the parameter is [decoded the size is checked](https://github.com/paritytech/substrate/blob/a47f200eebeb88a5bde6f1ed2be9728b82536dde/primitives/core/src/bounded/bounded_vec.rs#L279-L291). Consequently it is the same like including a `Vec` as a parameter, except the following:
1. We can use a custom error that can be pinned exactly to the location where it happened.
2. Using a `Vec` makes crafting the parameter easier for the application

I propose we use `Vec`, convert it in the dispatchable, create a custom error and create a new `Config` item, that contains the maximum length `MaxRejectReasonLen` and that can be queried by the app that crafts the transaction.